### PR TITLE
Math classes 1.0.7

### PR DESCRIPTION
**Pull request by Zimmi48**
###### Motivation for this change

Introduces compatibility with Coq 8.7.0.
Also adds the bignums package which is a dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Zimmi48** [comment]
Cc @jwiegley @siddharthist @vgbl @maximedenes.

File: pkgs/development/coq-modules/bignums/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+else with rev_and_sha."${coq.coq-version}";
+
+stdenv.mkDerivation rec {
+
+  name = "coq${coq.coq-version}-bignums";
+
+  src = fetchFromGitHub {
+    owner = "coq";
+    repo = "bignums";
+    inherit rev sha256;
```
Comment (siddharthist) [inline pkgs/development/coq-modules/bignums/default.nix]: Does this build? I don't see `rev` and `sha256` in the parent dictionary.

**Comment by siddharthist** [comment]
@Zimmi48 Thanks! Looking forward to having the new `bignums` in nixpkgs.

File: pkgs/development/coq-modules/bignums/default.nix (hunk 1 above)
Comment (Zimmi48) [reply pkgs/development/coq-modules/bignums/default.nix]: Yes it builds, the rev and sha256 are defined in an attribute set above that is imported using `with`.

File: pkgs/development/coq-modules/math-classes/default.nix [hunk 2]
```diff
@@ -1,21 +1,22 @@
-{ stdenv, fetchurl, coq }:
+{ stdenv, fetchFromGitHub, coq, coqPackages }:
 
-stdenv.mkDerivation {
-  name = "coq${coq.coq-version}-math-classes-1.0.6";
+if ! stdenv.lib.versionAtLeast coq.coq-version "8.6" then
+  throw "Math-Classes requires Coq 8.6 or later."
```
Comment (vbgl) [inline pkgs/development/coq-modules/math-classes/default.nix]: Is this true? How difficult is it to keep the old package for 8.5 alive?

File: pkgs/development/coq-modules/bignums/default.nix (hunk 1 above)
Comment (siddharthist) [reply pkgs/development/coq-modules/bignums/default.nix]: Ah, I see. The indentation is a little strange, it doesn't make it clear that the `mkDerivation` is a sub-expression of the if-else clause. Not a huge deal, though.

File: pkgs/development/coq-modules/math-classes/default.nix (hunk 2 above)
Comment (Zimmi48) [reply pkgs/development/coq-modules/math-classes/default.nix]: @vbgl There is no package for 8.5 in the sense that the previous package available in nixpkgs was already requiring Coq 8.6 (at least that's what the corresponding OPAM package says, I haven't checked that there was indeed a build failure).

Comment (vbgl) [reply pkgs/development/coq-modules/math-classes/default.nix]: My bad. You’re right: `coqPackages_8_5.math-classes` currently fails to build.

**Pull request merged**